services/slack/service.go

```diff
@@ -17,14 +19,21 @@ import (
 type Service struct {
 	configValue atomic.Value
 	logger      *log.Logger
+	client      *http.Client
 }
 
-func NewService(c Config, l *log.Logger) *Service {
+func NewService(c Config, l *log.Logger) (*Service, error) {
+	tlsConfig, err := getTLSConfig(c.SSLCA, c.SSLCert, c.SSLKey, c.InsecureSkipVerify)
+	if err != nil {
+		return nil, err
+	}
 	s := &Service{
 		logger: l,
+		client: &http.Client{Transport: &http.Transport{TLSClientConfig: tlsConfig}},
```

Comment:
The configuration can be changed at runtime via the API, there should be an Update method on the service which is in charge of applying new config changes. Can you update that method to allow for creating a new client when the TLS config changes. See the Alerta service for an example.